templates/default/consul-init.erb

```diff
@@ -37,6 +37,10 @@ is_running() {
     [ -f "$PIDFILE" ] && ps `get_pid` > /dev/null 2>&1
 }
 
+check_config() {
+    /usr/local/bin/consul configtest -config-dir /etc/consul.d
```

Comment:
Maybe you could use the same syntax that on line 23: 
```
CMD="<%= Chef::Consul.active_binary(node) %>
```

Rather than writing the absolute path of consul ?